server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -85,14 +86,17 @@ public QueryResource(
       QuerySegmentWalker texasRanger,
       ServiceEmitter emitter,
       RequestLogger requestLogger,
-      QueryManager queryManager
+      QueryManager queryManager,
+      InjectableValues guiceInjectableValues
   )
   {
     this.config = config;
     this.jsonMapper = jsonMapper.copy();
+    this.jsonMapper.setInjectableValues(guiceInjectableValues); // Workaround for https://github.com/FasterXML/jackson-databind/issues/696
```

Comment:
I agree with that. As a matter of fact, the only closable stream the objectMapper seems to be used on is the input stream where it parses the Query datatype. After trying a few quick examples it seems to function fine without the close. I'm going to try a few more tests then get rid of the copy completely.